scipy/interpolate/_ndbspline.py

```diff
@@ -0,0 +1,205 @@
+import operator
+import numpy as np
+
+from math import prod
+
+from . import _bspl  # type: ignore
+
+__all__ = ["NdBSpline"]
+
+
+def _get_dtype(dtype):
+    """Return np.complex128 for complex dtypes, np.float64 otherwise."""
+    if np.issubdtype(dtype, np.complexfloating):
+        return np.complex_
+    else:
+        return np.float_
+
+
+class NdBSpline:
+    """Tensor product spline object.
+
+    The value at point ``xp = (x1, x2, ..., xN)`` is evaluated as a linear
+    combination of products of one-dimensional b-splines in each of the ``N``
+    dimensions::
+
+       c[i1, i2, ..., iN] * B(x1; i1, t1) * B(x2; i2, t2) * ... * B(xN; iN, tN)
+
+
+    Here ``B(x; i, t)`` is the ``i``-th b-spline defined by the knot vector
+    ``t`` evaluated at ``x``.
+
+    Parameters
+    ----------
+    c : ndarray, shape (n1, n2, ..., nN, ...)
```

Comment:
The order of the arguments in this class's __init__ is t, c, k, so shouldn't the order of the arguments in this docstring be the same?